app/Http/Controllers/EventCheckoutController.php

```diff
@@ -819,4 +896,16 @@ public function showOrderTickets(Request $request, $order_reference)
         return view('Public.ViewEvent.Partials.PDFTicket', $data);
     }
 
+    public function showOrderDoc($folder_name, $file_name)
+    {
+        $fp = Storage::disk('local')->path('docs').'/'.$folder_name.'/'.$file_name;
+        if(is_file($fp)) {
+            return response()->file($fp);
+        }
+        else {
+            // Prevent bruteforce
+            sleep(1);
```

Comment:
```suggestion
```

https://www.php.net/manual/en/function.sleep.php

> Not really - the attacker could do 100 requests. Each request might take 2 seconds but it doesn't stop the number of requests done. You need to stop processing more than one request every 2 seconds rather than delay it by 2 seconds on each execution.

This can be done on the webserver e.g. nginx or using laravel's own Reatelimiting on the route:
https://laravel.com/docs/10.x/routing#rate-limiting